src/java.base/share/classes/jdk/internal/jrtfs/JrtFileSystem.java

```diff
@@ -175,7 +175,7 @@ public final String getSeparator() {
     @Override
     public PathMatcher getPathMatcher(String syntaxAndInput) {
         int pos = syntaxAndInput.indexOf(':');
-        if (pos <= 0 || pos == syntaxAndInput.length()) {
```

Comment:
Is this really a different bug? Should it be checking for `length() - 1` to throw IAE.
That would then throw if there was no string after the syntax selector;  for example "glob:"

Reply:
You have a point there. It does not want the ":" to be at the first position so it should not want it at the last position as the "pattern" part of the string would be empty. That fits with the specification
```
IllegalArgumentException.html - If the parameter does not take the form: syntax:pattern
```

Reply:
Corrected in 5765fa1c6883575da57499c7b70f6626443d869d.